src/test/java/org/terasology/logic/inventory/InventoryManagerTest.java

```diff
@@ -0,0 +1,207 @@
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.terasology.context.Context;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.moduletestingenvironment.ModuleTestingEnvironment;
+import org.terasology.world.block.BlockManager;
+import org.terasology.world.block.items.BlockItemFactory;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class InventoryManagerTest {
+
+    static final String URI_DIRT = "CoreAssets:Dirt";
+
+    private static ModuleTestingEnvironment context;
+
+    private Context hostContext;
+    private EntityManager entityManager;
+    private InventoryManager inventoryManager;
+    private BlockManager blockManager;
+    private EntityRef inventory;
+
+    @BeforeAll
+    public static void setup() throws Exception {
+        context = new ModuleTestingEnvironment() {
+            @Override
+            public Set<String> getDependencies() {
+                return Sets.newHashSet("CoreAssets", "Inventory");
+            }
+        };
+        context.setup();
+    }
+
+    @AfterAll
+    public static void tearDown() throws Exception {
+        context.tearDown();
+    }
+
+    @BeforeEach
+    public void beforeEach() {
+        hostContext = context.getHostContext();
+
+        entityManager = hostContext.get(EntityManager.class);
+        inventoryManager = hostContext.get(InventoryManager.class);
+        blockManager = hostContext.get(BlockManager.class);
+
+        inventory = entityManager.create(new InventoryComponent(3));
+    }
+
+    @ParameterizedTest
+    @ValueSource(bytes = {0, -1})
+    public void giveItem_zeroBlocks(int amount) {
+        EntityRef blockItem = getDirtBlockItem(entityManager, blockManager, amount);
+        inventoryManager.giveItem(inventory, EntityRef.NULL, blockItem);
+
+        final List<EntityRef> filledSlots = getFilledSlots(inventory);
+        Assert.assertEquals("No slots should be filled", 0, filledSlots.size());
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {1, 99})
+    public void giveItem_singleStack(int amount) {
+        EntityRef blockItem = getDirtBlockItem(entityManager, blockManager, amount);
+        inventoryManager.giveItem(inventory, EntityRef.NULL, blockItem);
+
+        final List<EntityRef> filledSlots = getFilledSlots(inventory);
+        Assert.assertEquals("Exactly one slot should be filled", 1, filledSlots.size());
+        filledSlots.forEach(item -> {
+            Assert.assertEquals("Slot should contain exactly the added block", blockItem, item);
+
+            final int maxStackSize = blockItem.getComponent(ItemComponent.class).maxStackSize;
+            Assert.assertEquals("Stack count should be " + amount + " or maximum stack size",
```

Comment:
I think in this test, the stack count should be exactly `amount` as you don't have any excess...